specs/phase1/beacon-chain.md

```diff
@@ -743,6 +405,9 @@ def process_operations(state: BeaconState, body: BeaconBlockBody) -> None:
     for_ops(body.attestations, process_attestation)
     for_ops(body.deposits, process_deposit)
     for_ops(body.voluntary_exits, process_voluntary_exit)
+    # Limit is dynamic based on active shard count
+    assert len(body.shard_headers) <= MAX_SHARD_HEADERS_PER_SHARD * get_active_shard_count(state, get_current_epoch(state))
```

Comment:
This doesn't protect from a shard having more headers then `MAX_SHARD_HEADERS`. If one shard has 5 and another has 3, they have 4 on average, which will pass the assertion.

Reply:
What's wrong with allowing one shard to have 5 and another 3? The normal case is 2 per shard; if there's an unexpected level of mischief (eg. a proposer making 10 malicious proposals and getting some attestations for each of them) on one shard, then why shouldn't all the slack be able to be directed toward that shard?

Reply:
I guess my confusion was caused by the name of the constant. To me it implies that it's the maximum number of headers a shard can have. I am not sure what would be a better name though...